src/components/common/SmartPrice.tsx

```diff
@@ -0,0 +1,65 @@
+import { calculatePrice, formatPrice, invertPrice, safeTokenName } from '@gnosis.pm/dex-js'
+import React, { useMemo, useState } from 'react'
+import BN from 'bn.js'
+import { TokenDetails } from 'types'
+
+import { SwapPrice } from 'components/common/SwapPrice'
+import { getMarket } from 'utils'
+
+interface Fraction {
+  denominator: BN
+  numerator: BN
+}
+
+interface Props {
+  buyToken: TokenDetails
+  sellToken: TokenDetails
+  price: Fraction
+}
+
+export const SmartPrice: React.FC<Props> = ({ buyToken, sellToken, price: priceFraction }) => {
+  const [isPriceInverted, setIsPriceInverted] = useState(false)
+  const { baseToken, quoteToken } = getMarket({ sellToken, receiveToken: buyToken })
+
+  const [price, priceInverse] = useMemo((): string[] => {
+    const buyOrderPrice = calculatePrice({
+      numerator: { amount: priceFraction.numerator, decimals: buyToken.decimals },
+      denominator: { amount: priceFraction.denominator, decimals: sellToken.decimals },
+    })
+    const buyOrderPriceInverse = invertPrice(buyOrderPrice)
+    const sellTokenIsQuote = buyToken === quoteToken
```

Comment:
? `sellTokenIsQuote` if `buyToken` = `quoteToken` doesn't make sense, should it be `const buyTokenIsQuote`

Reply:
hehe, yep, good catch. I guess I reversed the logic at some point and didn't update the var name